### ISSUE TITLE
Cannot compile using Xcode 12 beta

### ISSUE THREAD

**Issue by Ricardo1980** [urbanairship/ios-library#249]
### Report

Cannot compile using Xcode 12 beta
Version 12.0 beta (12A6159)

Right now my app has this in the pod file:
pod 'UrbanAirship-iOS-SDK', '12.1.2'

Which is old. I try to update it to 13.4.0 but version not found by cocoapods.

If I go to https://cocoapods.org/ and I enter UrbanAirship-iOS-SDK, the latest version is 12.1.2

Shouldn't be 13.4.0?

Anyway, if I use version 12.1.2 I receive this error:

Use of undeclared identifier '_style'; did you mean 'style'?

Line 93, this one:
_style = style;

File UAMessageCenterSplitViewController.m

And I cannot compile my app using Xcode 12 beta.

<img width="860" alt="Screenshot 2020-06-24 at 16 42 57" src="https://user-images.githubusercontent.com/738681/85590237-24ce1500-b63c-11ea-840e-8d29d65eda3a.png">

Any clue?

Thanks a lot.

### What Airship dependencies are you using?
I would say nothing, I am only using: pod 'UrbanAirship-iOS-SDK', '12.1.2'

### What are the versions of any relevant development tools you are using?
pod 'UrbanAirship-iOS-SDK', '12.1.2'
Version 12.0 beta (12A6159)

### What unexpected behavior are you seeing?
I cannot compile my app using Xcode 12 beta

### What is the expected behavior?
I should be able to compile my app using Xcode 12 beta

### What are the steps to reproduce the unexpected behavior?
Download Xcode 12 beta and try to compile your app

### Do you have logging for the issue?
Nope

**Comment by BrianBatchelder** [comment]
Our SDK changed in version 13. Please follow the [SDK 12 to 13 migration](https://github.com/urbanairship/ios-library/blob/master/Documentation/Migration/migration-guide-12-13.md) to fix your CocoaPods issue.

**Comment by JuhaO81** [comment]
There is now 'style' also in uisplitviewcontroller and that causes conflict

https://developer.apple.com/documentation/uikit/uisplitviewcontroller/3580915-style

**Comment by Ricardo1980** [comment]
I updated my code to 13.4.0 but I see the same issue, although different file.

UADefaultMessageCenterSplitViewController.m

<img width="830" alt="Screenshot 2020-06-25 at 10 17 39" src="https://user-images.githubusercontent.com/738681/85692318-37445f00-b6cd-11ea-87fa-059cd7affd77.png">

Can you check it? Thanks a lot!

**Comment by BrianBatchelder** [comment]
Thanks for the feedback. We have a fix for this issue in review and are currently working on a process to release it.

**Comment by scier** [comment]
The recent 13.5.0 release does not look like it addresses this issue; do you have a release plan for this fix?

This is the sole dependency in our app which does not build with the Xcode 12 beta, and this is actively interfering with our work on iOS 14 support, as this is important for debugging issues when running on iOS 14 -- which will become even more of an issue when iOS 14 usage increases with the forthcoming public beta.

**Comment by rlepinski** [comment]
@scier To fix the issue requires a major release since it's technically breaking an API. I am currently working on a beta release that you can use for now and a final major release in time for the latest iOS release.  Hope to have it out today

**Comment by scier** [comment]
Thanks for the insight -- that makes sense!

I'll put in a vote for a final major release as soon as possible, though, ideally well ahead of the iOS 14 release; a beta is appreciated as a workaround, but when this is included as a pod dependency-of-a-dependency, taking advantage of a beta version of a pod isn't always practical. Though I realize that doing a major release too early is also a risk you'd like to avoid, since who knows what *other* iOS API will change in iOS 14 between now and release; so understood it's a balancing act.

**Comment by scier** [comment]
I'm sure you've discussed the form of this breaking change internally, but just for another viewpoint, I wonder if you might consider either treating it as a non-breaking change gated by the SDK version and released as part of the 13.x series -- or a release to the 13.x to smooth the transition and allow us to build UA 13.x with the 14 beta SDK? E.g.

```
@property(nonatomic, strong) UAMessageCenterStyle *messageCenterStyle;
#if !defined(__IPHONE_14_0)
@property(nonatomic, strong) UAMessageCenterStyle *style;
#endif
```

```
#if !defined(__IPHONE_14_0)
- (void)setStyle:(UAMessageCenterStyle *)style {
    [self setMessageCenterStyle:style];
}
- (UAMessageCenterStyle *)style {
    return self.messageCenterStyle;
}
#endif
```

**Comment by rlepinski** [comment]
You know we have not, that is an interesting idea. I am not really a fan of this approach as someone might set the new view style property with wrong type, but it looks like the property is read only so assuming that doesn't change the compiler will fail. I will discuss with my team.

We are looking to do a major release that addresses this properly sometime early august, then hopefully a minor release again with any new iOS 14 functionality when Xcdoe 12 GM is released.

**Comment by tapi** [comment]
I was able to fix the compiler issue locally by just explicitly declaring the backing ivar, this is what my modified `.m` looks like

```objective-c
@implementation UAMessageCenterSplitViewController {
	UAMessageCenterStyle * _style;
}
```

**Comment by rlepinski** [comment]
We released a beta where we renamed it from `style` to `messageCenterStyle`. Going to do the macro rename suggested by @scier for 13.x sometime next week, but that only applies to carthage and pods. 

@tapi Thanks for the work around. Are you using message center? I am not sure how the OS would interact with the `style` property if it accesses the getter that we now have overridden.

**Comment by tapi** [comment]
We aren't using the message center so I don't know what interactions would result from just this change but I think synthesized accessors should function just fine, if not you could still implement the getter and avoid the breaking API change

**Comment by scier** [comment]
Thanks for taking the extra time to help smooth this process out and make our lives easier @rlepinski! Much appreciated :-)

**Comment by marc-scig** [comment]
@tapi We considering doing something like that for a moment as well, but it's dangerous – overriding the property means that if the class is upcast the return value will be typed as a (garbage) UISplitViewControllerStyle value, and could cause hard to diagnose bugs. The safest thing is either to rename the property or do the thing @scier suggested and conditionally compile it in. We should have a patch on 13.x available soon.

**Comment by marc-scig** [comment]
@scier @tapi: 13.5.1 is out. See the CHANGELOG for more details.

**Comment by srinadh-k** [comment]
Hi we are getting following compilation error on simulator. But working fine on real device. any idea?

`Library not found for -lAirshipExtensions
`

**Comment by rlepinski** [comment]
@srinadh-k please open a new issue. Tell us how you are integrating the SDK, Xcode version, and SDK version. Thanks!
